src/unwinder/arch/aarch64.rs

```diff
@@ -91,30 +87,33 @@ pub extern "C-unwind" fn save_context(f: extern "C" fn(&mut Context, *mut ()), p
             ",
             options(noreturn)
         );
-    }
+    };
+    (maybesavefp(fp)) => {
+        "
+        stp d8, d9, [sp, 0x140]
+        stp d10, d11, [sp, 0x150]
+        stp d12, d13, [sp, 0x160]
+        stp d14, d15, [sp, 0x170]
+        "
+    };
+    (maybesavefp()) => { "" };
 }
 
-pub unsafe fn restore_context(ctx: &Context) -> ! {
+#[naked]
+pub extern "C-unwind" fn save_context(f: extern "C" fn(&mut Context, *mut ()), ptr: *mut ()) {
     unsafe {
+        #[cfg(target_feature = "fp-armv8")]
```

Comment:
AFAIK there is no such a target feature in aarch64 (at least on the Rust side):

```console
$ rustc +nightly --print cfg --target aarch64-unknown-linux-gnu | grep target_feature
target_feature="neon"
$ rustc +nightly --print cfg --target aarch64-unknown-none | grep target_feature
target_feature="neon"
$ rustc +nightly --print cfg --target aarch64-unknown-none-softfloat | grep target_feature
```

https://github.com/rust-lang/rust/blob/b5b13568fb5da4ac988bde370008d6134d3dfe6c/compiler/rustc_target/src/target_features.rs#L87

arm has it though: https://github.com/rust-lang/rust/blob/b5b13568fb5da4ac988bde370008d6134d3dfe6c/compiler/rustc_target/src/target_features.rs#L62

Reply:
Oh, weird—the error when I tried to use `unwinding` mentioned `fp-armv8` explicitly. I'll fix that.

Reply:
To be more clear: when I tried to use the current `unwinding` master branch on `aarch64-unknown-none-softfloat`, I got a compile error saying that the registers `d[n]` require target feature `fp-armv8`. This is likely a diagnostic issue, and I'll see if I can reproduce it to send that over to rustc.